src/tests/itests/xi_itest_update.c

```diff
@@ -0,0 +1,91 @@
+/* Copyright (c) 2003-2017, LogMeIn, Inc. All rights reserved.
+ *
+ * This is part of the Xively C Client library,
+ * it is licensed under the BSD 3-Clause license.
+ */
+
+#include <xi_itest_update.h>
+#include "xi_itest_helpers.h"
+#include "xi_backoff_status_api.h"
+
+#include "xi_debug.h"
+#include "xi_globals.h"
+#include "xi_handle.h"
+
+#include "xi_memory_checks.h"
+#include "xi_itest_layerchain_ct_ml_mc.h"
+#include "xi_itest_mock_broker_layerchain.h"
+
+/* Depends on the xi_itest_tls_error.c */
+extern xi_context_t* xi_context;
+extern xi_context_handle_t xi_context_handle;
+extern xi_context_t* xi_context_mockbroker;
+/* end of dependency */
+
+int xi_itest_update_setup( void** fixture_void )
+{
+    /* clear the external dependencies */
+    xi_context            = NULL;
+    xi_context_handle     = XI_INVALID_CONTEXT_HANDLE;
+    xi_context_mockbroker = NULL;
+
+    xi_memory_limiter_tearup();
+
+    *fixture_void = NULL;
+    // *fixture_void = xi_itest_connect_error__generate_fixture();
+
```

Comment:
This test does nothing now. This is the point I decided to put this to review since it grew too big. I will continue from here. This will change soon but for now I'd leave as it is if possible.